server/scheduler.py

```diff
@@ -79,8 +79,14 @@
 
     if request_recv_socket in sockets and sockets[request_recv_socket] == zmq.POLLIN:
         serialized = request_recv_socket.recv()
+        try:
+            request = pickle.loads(serialized)
+            logging.info(f"Received request: {request}")
+        except (pickle.UnpicklingError, KeyError) as e:
+            logging.info(f"Exception of type {type(e).__name__} occurred with request: {serialized}")
+            continue
         logging.info(f"Received request {serialized}")
-        worker = scheduler.schedule_request(serialized)
+        worker = scheduler.schedule_request(request)
         if worker:
```

Comment:
Can you revert the changes in scheduler for now? It'll be cleaner to put those in another PR